File: WebApplication1/WebApplication1/Models/Report_Log.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace WebApplication1.Models
{
    public class Report_Log
    {
        [Key]
        public int Id { get; set; }
        
        public int User_Id { get; set; }

        public int Reported_User_Id { get; set; }

        public string Message { get; set; }
    }
}

File: Angular/src/app/models/ReportLog.ts
export class ReportLog {
    id: number;
    userId: number;
    reportedUserId: number;
    message: string;
}

export class CreateReportDTO {
    requestorId: number;
    targetId: number;
    message: string;
}

export class UserReportDTO {
    userId: number;
    reportId: number;
    until: Date;
}